perf(ivy): chain host binding instructions

crisbeto opened the pull request:
Adds chaining to the `property`, `attribute` and `updateSyntheticHostBinding` instructions when they're used in a host binding.

This PR resolves FW-1404.

pkozlowski-opensource commented on packages/compiler-cli/test/compliance/r3_view_compiler_binding_spec.ts:
```diff
@@ -860,6 +860,251 @@ describe('compiler compliance: bindings', () => {
+      expectEmit(result.source, template, 'Incorrect template');
+    });
+
+    it('should chain multiple host attribute bindings in the presence of other instructions',
```
nit suggestion: "other instructions" -> "other bindings"

pkozlowski-opensource reviewed (approved):
LGTM, "compiler people" should still have a look :-)

crisbeto committed:
perf(ivy): chain host binding instructions

Adds chaining to the `property`, `attribute` and `updateSyntheticHostBinding` instructions when they're used in a host binding.

This PR resolves FW-1404.

kara reviewed (approved):
LGTM

kara commented:
[VE presubmit](http://test/OCL:255531668:BASE:255531678:1561692911291:9b406e21)

[Ivy presubmit](http://test/OCL:255531668:BASE:255532997:1561693800240:1ecc35b3)

kara commented:
merge-assistance: global
